File: FE/OnlineEvent/src/app/types/event.ts
export interface EventPayload {
  Nombre: string;
  Fecha: string;
  Lugar: string;
  Estado: string;
};

// Request para crear un evento
export interface CreateEventRequest {
  name: string;
  date: string;      // ISO string: 2026-01-20
  location?: string;
}

// Response del backend
export interface EventResponse {
  id: string;
  name: string;
  date: string;
  location?: string;
  createdBy: string;
  createdAt: string;
}

// Modelo base (opcional, reutilizable)
export interface Event {
  id: string;
  name: string;
  date: string;
  location?: string;
}

File: BE/mic-event-api/Domain/Entities/Event.cs
namespace mic_event_api.Domain.Entities
{
    public class Event
    {
        public Guid Id { get; set; }
        public string Nombre { get; set; } = string.Empty;
        public string Fecha { get; set; } = string.Empty;
        public string Lugar { get; set; } = string.Empty;

        public string Estado { get; set; } = string.Empty;

        public Event(Guid id, string nombre, string fecha, string lugar, string estado)
        {
            Id = id;
            Nombre = nombre;
            Fecha = fecha;
            Lugar = lugar;
            Estado = estado;
        }
    }
}
